### DISCUSSION TITLE
Is DoQ from Android with Adguard Home possible?

### DISCUSSION THREAD

**Discussion by pwsnla** [AdguardTeam/AdGuardHome#6005] [Q&A]
Android apparently supports QUIC protocol since Android 11. From reading it appears that initially only Google DNS and Cloudflare DNS were supported. Is this still the case? 

I have enabled and configured DoQ in Adguard Home but I can only get DoT working from Android. Can anyone shed any light on this?

**Comment by lucasmz-dev** [comment]
Android DOES NOT support DoQ.
It supports DNS over HTTP/3. (DNS over HTTPS using HTTP/3)

These are different, they use different ports, and do not work with one another, they are however to ways to use QUIC to deliver DNS. Unfortunately, DNS over HTTP/3 is not as efficient, it is more supported because it's just DNS over HTTPS using QUIC. Pretty much the only public DNS server out there that supports DoQ unfortunately, is Adguard.

And even for HTTP/3, it only works for two domains, `dns.google` and `cloudflare-dns.com`, `one.one.one.one` won't work and will use DNS over TLS.
It also assumes that the server uses `/dns-query` and not something else.
